internal/terraform/node_resource_abstract_instance.go

```diff
@@ -584,8 +584,9 @@ func (n *NodeAbstractResourceInstance) writeChange(ctx EvalContext, change *plan
 }
 
 // refresh does a refresh for a resource
-func (n *NodeAbstractResourceInstance) refresh(ctx EvalContext, deposedKey states.DeposedKey, state *states.ResourceInstanceObject) (*states.ResourceInstanceObject, tfdiags.Diagnostics) {
+func (n *NodeAbstractResourceInstance) refresh(ctx EvalContext, deposedKey states.DeposedKey, state *states.ResourceInstanceObject) (*states.ResourceInstanceObject, tfdiags.Diagnostics, *providers.Deferred) {
```

Comment:
nit: style preference, in go we typically always put the `error` last if we're returning it. In this case the diagnostics are behaving as the error, so I'd have `(*states.ResourceInstanceObject, *providers.Deferred, tfdiags.Diagnostics)` as the order of returned objects.